plugins/org.eclipse.xtext.xbase.ui/src/org/eclipse/xtext/xbase/ui/refactoring/NewFeatureNameUtil.java

```diff
@@ -121,10 +121,18 @@ protected String getFancyDefaultName(XExpression expression) {
 		if(expression instanceof XClosure)
 			return "function";
 		LightweightTypeReference type = batchTypeResolver.resolveTypes(expression).getActualType(expression);
-		if (type != null) {
-			if (type.isPrimitive())
+		return toVariableName(type);
+	}
+	
+	protected String toVariableName(LightweightTypeReference type) {
+		if(type != null) {
+			if(type.isPrimitive())
 				return type.getSimpleName().substring(0, 1);
-			else
+			else if(type.isArray()) 
+				return toVariableName(type.getComponentType()) + 's';
```

Comment:
cosmetic: two dimensional arrays would get more than one 's' in the end, simple names with 's' and the end would be strange, too